video/out/vulkan/utils.c

```diff
@@ -496,6 +518,10 @@ bool mpvk_device_init(struct mpvk_ctx *vk, struct mpvk_device_opts opts)
     const char **exts = NULL;
     int num_exts = 0;
     MP_TARRAY_APPEND(tmp, exts, num_exts, VK_KHR_SWAPCHAIN_EXTENSION_NAME);
+#if HAVE_WAYLAND || HAVE_X11
+    MP_TARRAY_APPEND(tmp, exts, num_exts, VK_KHR_EXTERNAL_MEMORY_EXTENSION_NAME);
+    MP_TARRAY_APPEND(tmp, exts, num_exts, VK_KHR_EXTERNAL_MEMORY_FD_EXTENSION_NAME);
+#endif
```

Comment:
It's not okay to unconditionally require this extension being available, even if you guard it behind HAVE_WAYLAND / HAVE_X11. Furthermore, that guard should not be necessary in theory.

Fix both problems by checking for availability of the extensions you need, and then making the feature dependent on that capability instead.

Reply:
I have adjusted this to do a proper availability check for the extensions.